### PR TITLE
feat: [PIE-2510]: Removing merge logic out of PlanCreationResponse

### PR DESCRIPTION
**Pull request by prashantsharma04**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30381)

### CODE REVIEW

File: 882-pms-sdk-core/src/main/java/io/harness/pms/sdk/core/plan/creation/beans/MergePlanCreationResponse.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  @Singular Map<String, PlanNode> nodes;
+  Dependencies dependencies;
+  YamlUpdates yamlUpdates;
+  @Singular("contextMap") Map<String, PlanCreationContextValue> contextMap;
+  GraphLayoutResponse graphLayoutResponse;
+
+  String startingNodeId;
+  @Singular List<String> errorMessages;
+
+  public void merge(PlanCreationResponse other) {
```
Comment (vaibhav-si) [inline 882-pms-sdk-core/src/main/java/io/harness/pms/sdk/core/plan/creation/beans/MergePlanCreationResponse.java:41]: rename other

**Comment by prashantsharma04** [comment]
trigger messagecheck

**Review by prashantsharma04 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30381)* status: 7 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[882-pms-sdk-core/src/main/java/io/harness/pms/sdk/core/plan/creation/beans/MergePlanCreationResponse.java, line 41 at r1](https://reviewable.io/reviews/harness/harness-core/30381#-MujnK5mNb1-303rt2rh-r1-41:-Mut-U4ZCCeuo7FKpe8f:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3a1b1feaef5ade0afe260f3a3274017b412cf1ad/882-pms-sdk-core/src/main/java/io/harness/pms/sdk/core/plan/creation/beans/MergePlanCreationResponse.java#L41)):*
<details><summary><i>Previously, vaibhav-si (Vaibhav Singhal) wrote…</i></summary><blockquote>

rename other
</blockquote></details>

Done.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 3 of 7 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30381)* status: 7 file(s) to review. 1 of 2 approvals obtained.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 4 of 7 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30381)* status: :shipit: complete! ✓

**Pull request merged**